### ISSUE TITLE
I use the transformer of YOLOv5,but after 11 epochs the model did't to train.

### ISSUE THREAD

**Issue by RXZ-BUPT** [ultralytics/yolov5#8421]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I use this order to train the model :
`python train.py --cfg models/hub/yolov5s-transformer.yaml --data data/train.yaml --hyp hyps/hyp.scratch-low.yaml --batch-size -1 --device 0,1 --optimizer Adam`
but the result is not good :
 Epoch   gpu_mem       box       obj       cls    labels  img_size
    36/299     11.7G       nan       nan       nan       180       640: 100%|███
               Class     Images     Labels          P          R     mAP@.5 mAP@
                 all         73          0          0          0          0          0

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    37/299     11.7G       nan       nan       nan       246       640: 100%|███
               Class     Images     Labels          P          R     mAP@.5 mAP@
                 all         73          0          0          0          0          0

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    38/299     11.7G       nan       nan       nan       283       640: 100%|███
               Class     Images     Labels          P          R     mAP@.5 mAP@
                 all         73          0          0          0          0

**Comment by AyushExel** [comment]
@RXZ-BUPT Have you tried training on non-transformers variants. In your console log, all train losses are nans so there might be a problem with th
e training data. On training on the official colab, I don't see nans, but yes the mAPs are 0 because it is training from scratch on toy dataset
<img width="1175" alt="Screenshot 2022-07-01 at 7 52 25 AM" src="https://user-images.githubusercontent.com/15766192/176810996-f45f014d-4134-491a-b7e8-d25acda880e2.png">

**Comment by RXZ-BUPT** [comment]
@AyushExel Thank you to answer my question. I have tried training on non-transformer, for example,the yolov5x.yaml.And i didn't have this problem.But when I used the yolov5s-transformer.yaml,I would get this problem.Why?

**Comment by ismh16** [comment]
I have the same situation.I guess the problem is in /models/transformer.yaml and /models/yolo.py.
Wish author can provide a demonstration to help us . thanks

**Comment by glenn-jocher** [comment]
@ismh16 the nan losses may indicate instability in your training. To solve this, ensure the transformer model has enough capacity to capture the complexity of your dataset. Incrementally increase model capacity and use a smaller learning rate. You might also want to check for data quality issues, such as invalid bounding boxes. If the problem persists, please share more details about your dataset and hyperparameters for further assistance. Thank you!
